packages/token-bridge-sdk/src/hooks/useArbTokenBridge.ts

```diff
@@ -1014,6 +1026,81 @@ export const useArbTokenBridge = (
     )
   }
 
+  // mutates provided array - assigns deadlineBlockNumbers to all asserted but unconfimred l2tol1 messages 
+  const addNodeDeadlineToUnconfirmedWithdrawals = async(l2ToL1Data: L2ToL1EventResultPlus[])=>{
+    if(l2ToL1Data.length === 0) return []
+    const l1NetworkID = await l1NetworkIDCached()
+    const oldOutboxOffset = (()=>{
+      switch (l1NetworkID) {
+        case '1':
+          return 30
+        case '4':
+          return 326
+        default:
+          throw new Error(`Unrecognized L1 network ${l1NetworkID}`)
+      }
+    })()
+
+    // get smallest batch number in messages for lower bound on graph query    
+    const smallestBatchNumber = l2ToL1Data.reduce((acc: BigNumber, currentL2ToL1Data:L2ToL1EventResultPlus )=>{
+      return acc.lt(currentL2ToL1Data.batchNumber) ? acc : currentL2ToL1Data.batchNumber
+    }, l2ToL1Data[0].batchNumber).toNumber()    
+    const nodes = await getNodes(l1NetworkID, smallestBatchNumber)
+  
+
+    const unconfirmedWithdrawals = l2ToL1Data.filter((l2ToL1Data)=> l2ToL1Data.outgoingMessageState !==  OutgoingMessageState.CONFIRMED && l2ToL1Data.outgoingMessageState !==  OutgoingMessageState.EXECUTED ).sort((msgA,msgB)=>{
+      // ensure sorted in ascending order by timestamp
+      return +msgA.timestamp - +msgB.timestamp
+    })
+    
+    let currentNodeIndex = 0
+    let currentNode: NodeDataResult|undefined = nodes[currentNodeIndex]
+    // get node ids for messages included in a node, preserving order of unconfirmedWithdrawals array
+    const nodeIDs = unconfirmedWithdrawals.map((l1ToL2Data: L2ToL1EventResultPlus)=>{
+
+        const batchNumberWithOffset = l1ToL2Data.batchNumber.toNumber() + oldOutboxOffset
+        
+        // find first node with aftersendCount >= current messages batch number ('afterSendCount' is batchcount )
+        while(currentNode && +currentNode.afterSendCount < batchNumberWithOffset ) {
+          currentNodeIndex++
+          currentNode = nodes[currentNodeIndex]
+        }
+        // if we've reached the end of the node array, the message hasn't been included in a node, (so undefined)
+        return currentNode ? BigNumber.from(currentNode.id) : null
+      }).filter(notNull)
+      
+      const rollupAddress  = bridge.l1Bridge.network.ethBridge?.rollup
+      if(!rollupAddress) throw new Error("Could not get rollup address")
+      const rollupIface = Rollup__factory.createInterface()
```

Comment:
Not to address in this PR but I think we only use interfaces from typechain, yet the factories also include the contract bytecode which is probably bloating the package size when not needed